### PR TITLE
Ftrack api

### PR DESCRIPTION
**Pull request by tokejepsen**
This is my suggestion for a workflow with the new api. This also addresses some problems with the old workflow, so now there are only a collector and integrator plugin.

The two workflows are separated, so both can be used at the same time and also provides backwards compatibility.

Have a look at the [usage](https://github.com/tokejepsen/pyblish-ftrack/blob/f9d5c7b02607ed86ded0532136b4c743581534bf/README.md#new-api) documentation for the details.

One thing I not certain about is providing defaults for a succesfull publish. On one side it makes it super simple to use and get started with, but on the other side it might also be too magical for new users. One the discussion :smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,40 +4,128 @@
...
 To succesfully publish an instance to ftrack using this extension you need to append a few data members to the instance you want to publish.
 
-**Required data:**
+**Required data**
 
-- ftrackComponents - Dictionary of dictionaries where each key is a component name (Instance data)
- - Each component (key) should be a dictionary that can have 2 keys
-   - *path* (required): holds path to the file being published
-    - *reviewable* (optional): tells extension whether you want component to be uploaded to ftrack servers and made available for webReview
+- ftrackComponents (dict) - Dictionary of dictionaries where each key is a component name. There are required and optional keys:
```
Comment (mottosso) [inline README.md:95]: Have a look at whether a table is a good fit for this kind of data. I find it useful for two-dimensional data like this.

[Example](https://github.com/mottosso/Qt.py#documentation)

**Review by mottosso — APPROVED** [review]
I can't review code quality without access to ftrack, and I'm not sure what else to add except that it looks good to me.

File: README.md (hunk 1 above)
Comment (tokejepsen) [reply README.md:95]: Sorry, not sure what you mean by a table? A list of dictionaries?

Comment (mottosso) [reply README.md:95]: | I   | Mean
|:---|:-----
| A  | Table

Have a look at the example.

**Comment by tokejepsen** [comment]
@mkolar what do you think about this?

**Comment by mkolar** [comment]
To be honest I don't have the dev time to go through it properly now. I was however planning to transition us to the new API for quite some time ad I trust your judgement on the implementation here.

In regards to the default values. I'd certainly keep them there (the ones you put in the readme make sense to me). It's simple enough not to appear as black magic behind the curtains and it allow people to be up and running to a certain degree right away. At the same time it does a good job serving as an example of what the data should look like, so let's keep it please.

**Comment by tokejepsen** [comment]
Should I merge then?

Have been working with this new workflow for a little while now, and its much more flexible than the old api. Can even use it in Hiero.

**Comment by mkolar** [comment]
yeah go for it. I'll pull it from the master then and give it a shot.

**Comment by tokejepsen** [comment]
Done :)

**Pull request merged**